app/models/spree/payment.rb

```diff
@@ -47,6 +47,7 @@ class Payment < ApplicationRecord
     scope :with_state, ->(s) { where(state: s.to_s) }
     scope :completed, -> { with_state('completed') }
     scope :incomplete, -> { where(state: %w(checkout pending requires_authorization)) }
+    scope :checkout, -> { with_state('checkout') }
```

Comment:
Ideally you would have added a unit test for this in the same commit. But it's so simple that it's only a bonus.

Reply:
Was thinking about it but then questioned the benefits vs cost of the unit test. How do you usually test these on ofn? Would you actually create payments with different states and then evaluate the result of the scoped query?

Reply:
Yes, exactly. If the `with_state` scope was already covered by a unit test then it would be less important. In the end, your other specs are covering this, too. So I don't mind you not adding a spec for this. I just wanted to let you know that when I'm feeling like I want to do everything perfect then I would add a spec there. I would create at least two records and make sure that the right one is returned.

Reply:
Got it, thanks @mkllnk!